openmc/mgxs/mgxs.py

```diff
@@ -3154,30 +3154,133 @@ def xs_tally(self):
                 raise ValueError(msg)
 
             # Switch EnergyoutFilter to EnergyFilter
+            if 'scatter-1' in self.tallies:
+                p1_tally = self.tallies['scatter-1']
+                old_filt = p1_tally.filters[-2]
+                new_filt = openmc.EnergyFilter(old_filt.values)
+                p1_tally.filters[-2] = new_filt
+
+                p1_tally = p1_tally.get_slice(filters=[openmc.LegendreFilter], 
+                    filter_bins=[('P1',)],squeeze=True)
+                p1_tally._scores = ['scatter-1']
+                total_xs = self.tallies['total'] / self.tallies['flux (tracklength)']
+                trans_corr = p1_tally / self.tallies['flux (analog)']
+                transport = total_xs - trans_corr
+                diff_coef = transport**(-1) / 3.0
+                self._xs_tally = diff_coef
+                self._compute_xs()
+
+            else:
+                self._xs_tally = self.tallies[self._rxn_type] / self.tallies['flux (tracklength)']
+                self._compute_xs()
+
+        return self._xs_tally
+
+    def get_condensed_xs(self, coarse_groups, condense_diff_coef=True):
+        """Construct an energy-condensed version of this cross section.
+
+        Parameters
+        ----------
+        coarse_groups : openmc.mgxs.EnergyGroups
+            The coarse energy group structure of interest
+
+        Returns
+        -------
+        MGXS
+            A new MGXS condensed to the group structure of interest
+
+        """
+
+        cv.check_type('coarse_groups', coarse_groups, EnergyGroups)
+        cv.check_less_than('coarse groups', coarse_groups.num_groups,
+                           self.num_groups, equality=True)
+        cv.check_value('upper coarse energy', coarse_groups.group_edges[-1],
+                       [self.energy_groups.group_edges[-1]])
+        cv.check_value('lower coarse energy', coarse_groups.group_edges[0],
+                       [self.energy_groups.group_edges[0]])
+
+        # Clone this MGXS to initialize the condensed version
+        condensed_xs = copy.deepcopy(self)
+
+        if condense_diff_coef:
             p1_tally = self.tallies['scatter-1']
```

Comment:
Would this not fail if the user tried condensing an MGXS object already condensed? (back to the above: `self.tallies['scatter-1']` wouldnt be present.